### PR TITLE
Feature/xcframeworks binary download preparing refactor

### PR DESCRIPTION
**Pull request by michallaskowski**
As discussed in #2925
The refactor in itself gives this project and me nothing, but should enable easier addition of features (in my case I need binary xcframeworks) - by modularization and clear(er) set of responsibilities => allowing to unit tests particular classes => avoiding bugs and helping to understand the code faster. (In any particular order really, 3am for me now)

From Project.swift I carved out ~450 lines of code into separate classes
- BinaryInstaller
- BinaryFrameworkDownloader - download binary frameworks. Could be further split into GitHub downloader (or extension on Tentacle.Client) and the rest.
- FrameworkInformationProvider - provides info about frameworks already on the disk.

### CODE REVIEW

File: Source/CarthageKit/Project.swift [hunk 1]
```diff
@@ -106,23 +106,38 @@ public final class Project { // swiftlint:disable:this type_body_length
...
+        set {
+            binaryInstaller.useNetrc = newValue
+        }
+    }
 
 	/// Sends each event that occurs to a project underneath the receiver (or
 	/// the receiver itself).
-	public let projectEvents: Signal<ProjectEvent, NoError>
-	private let _projectEventsObserver: Signal<ProjectEvent, NoError>.Observer
+    public var projectEvents: Signal<ProjectEvent, NoError>
```
Comment (tmspzz) [inline Source/CarthageKit/Project.swift:120]: private (set) ?

**Comment by tmspzz** [comment]
Thanks for the PR and the refactoring work! This will certainly help going forward. I left one comment.

File: Source/CarthageKit/Project.swift (hunk 1 above)
Comment (michallaskowski) [reply Source/CarthageKit/Project.swift:120]: hm, I think it is some leftover, it can be `let` as it was.

**Comment by DavidBrunow** [comment]
I think all the new indentation uses spaces while the rest of Carthage uses tabs.

**Comment by michallaskowski** [comment]
Yeah, I also noticed after the opening a PR. It is the only project so far I have seen with tabs, I am searching how to change Xcode to use tabs, per project, not for all of them.

**Comment by michallaskowski** [comment]
@tmspzz thanks for the approval. What happens now? Is there a required set of approvers? I see some other PRs that are approved but sitting in open state, so best to be sure.
If this is merged I will prepare a draft with Readme updated, and then probably would start picking commits from your work and writing tests for binary xcframeworks. wdyt?

**Comment by tmspzz** [comment]
@michallaskowski it would actually be better to merge this into my fork to minimize risk for the next release. For the steps after that it's best if we continue discussion on Twitter

**Comment by jdhealy** [comment]
Appreciate you crafting on this @michallaskowski ☀️👍

**Comment by tmspzz** [comment]
@michallaskowski changed destination branch from master to feature branch

**Pull request merged**